### PR TITLE
Add reusable workflow for Java instrumentation tests

### PR DESCRIPTION
**Pull request by twcrone**
### Overview
Trying out reusable workflow for Java instrumentation tests

### CODE REVIEW

**Review by XiXiaPdx — APPROVED** [review]
LGTM!

File: .github/workflows/X-Reusable-Test.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      # Check ENV variables
+      - name: Check environmental variables
+        run: printenv | sort -f
+
+      ## AWS jars - plan to cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.aws-access-key-id }}
+          aws-secret-access-key: ${{ secrets.aws-secret-access-key }}
```
Comment (meiao) [inline .github/workflows/X-Reusable-Test.yml:82]: Can't the reusable workflow access the repo secrets directly?
That would save a few lines in the concrete workflows.

Comment (twcrone) [reply .github/workflows/X-Reusable-Test.yml:82]: It failed when accessing directly from reusable.

File: .github/workflows/X-Reusable-Test.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
+name: X - Reusable Test Workflow
```
[the remark this answers is not in the document]
Comment (twcrone) [reply .github/workflows/X-Reusable-Test.yml:1]: If we keep it, we can rename to something appropriate, but we may need only one reusable workflow and perhaps not one specific to instrumentation.  We shall see.

**Pull request merged**